internal/pagination/pagination.go

```diff
@@ -0,0 +1,53 @@
+package pagination
+
+import (
+	"context"
+	"strconv"
+
+	"github.com/rs/zerolog"
+)
+
+type ctxKeyType int
+
+const (
+	defaultValueLimit             = 100
+	defaultValueOffset            = 0
+	offsetCtxKey       ctxKeyType = iota
+	limitCtxKey
+)
+
+// WithOffset returns context copy with offset value.
+func WithOffset(ctx context.Context, offsetStr string) context.Context {
+	logger := zerolog.Ctx(ctx)
+	offset, err := strconv.Atoi(offsetStr)
+	if err != nil || offset < 0 {
+		logger.Trace().Msg("Offset is missing, setting offset value to 0")
```

Comment:
This log will swallow the reason (the error message) in case user provides an invalid input (e.g. letters instead numbers). I suggest to use `Err` function and rephrase the message to "Offset missing or invalid, setting...".

The same problem below.

Reply:
Do you mean like that? 
`logger.Err(err).Msg("Offset is missing or invalid, setting offset value to 0")
`

Reply:
Exactly.